helper/keysutil/policy.go

```diff
@@ -429,7 +451,26 @@ func (p *Policy) NeedsUpgrade() bool {
 	return false
 }
 
-func (p *Policy) Upgrade(ctx context.Context, storage logical.Storage) error {
+func (p *Policy) Upgrade(ctx context.Context, storage logical.Storage) (retErr error) {
+	priorKey := p.Key
+	priorLatestVersion := p.LatestVersion
+	priorMinDecryptionVersion := p.MinDecryptionVersion
+	priorConvergentVersion := p.ConvergentVersion
+	priorKeys, err := copystructure.Copy(p.Keys)
```

Comment:
Is there a reason for using copystructure here vs doing a manual copy like in the Persist() and Rotate() functions? The only difference I see is that of the `nil` check on `p.Keys`.